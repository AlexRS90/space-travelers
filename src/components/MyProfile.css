.table-wrapper {
  display: flex;
  justify-content: center;
  gap: 20px;
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.mission-section {
  width: 30%;
}

.rockets-wrapper {
  width: 35%;
  padding: 10px;
}

.rockets-wrapper h2 {
  color: #314e72;
  text-align: center;
}

.rockets-wrapper h3 {
  padding: 5px;
  color: #314e72;
}

.gap {
  display: flex;
  flex-direction: column;
}

.info-rocket {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-radius: 10px;
}

.info-rocket a {
  color: #314e72;
}

.info-rocket:nth-child(odd) {
  background-color: #bdc4cc;
}
