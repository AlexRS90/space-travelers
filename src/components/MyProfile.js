/* eslint-disable max-len */
import React, { } from 'react';
import { useSelector } from 'react-redux';
import MissionCardProfile from './MissionCardProfile';

const MyProfile = () => {
  const missions = useSelector((state) => state.missions);
  const reservedMissions = missions.filter((mission) => mission.reserved === true);

  return (
    <>
      {reservedMissions && reservedMissions.map((reserMisson) => <MissionCardProfile reserMisson={reserMisson} key={reserMisson.mission_id} />)}
    </>
  );
};

export default MyProfile;
