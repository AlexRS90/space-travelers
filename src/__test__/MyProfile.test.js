import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyProfile from '../components/MyProfile';

afterEach(() => {
  cleanup();
});

describe('render profile to the Dom', () => {
  test('render myprofile to the dom', async () => {
    render(<Provider store={store}><MyProfile /></Provider>);
    const myProfileElelement = await screen.findByTestId('profile-page');
    expect(myProfileElelement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Provider store={store}><MyProfile /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
