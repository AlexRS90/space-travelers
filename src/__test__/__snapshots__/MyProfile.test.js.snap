// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render profile to the Dom matches snapshot 1`] = `
<div
  className="table-wrapper"
  data-testid="profile-page"
>
  <table
    className="table table-striped mission-section"
  >
    <h2>
      No Missions booked Yet
    </h2>
  </table>
  <div
    className="rockets-wrapper"
  >
    <h2>
      No Rockets Reserved Yet
    </h2>
    <div
      className="gap"
    />
  </div>
</div>
`;
